openff/bespokefit/bespoke/smirks.py

```diff
@@ -158,148 +180,232 @@ def _get_all_bespoke_smirks(
         values.
         """
         bespoke_smirks = []
+        fragment_is_parent = True
+        fragment_molecule = fragment_data.fragment_molecule
+        # check if the fragment is the same as the parent, do it once for speed
+        if fragment_data.parent_molecule != fragment_molecule:
+            fragment_is_parent = False
```

Comment:
```suggestion
        fragment_molecule = fragment_data.fragment_molecule
        # check if the fragment is the same as the parent, do it once for speed
        fragment_is_parent = fragment_data.parent_molecule == fragment_molecule
```

Reply:
Nice, I would never have thought of that!